src/main/java/games/strategy/util/Util.java

```diff
@@ -122,4 +130,19 @@ public static String getStringFromInputStream(InputStream in) {
     }
     return builder.toString();
   }
+
+  /**
+   * Creates a hash of the given String based on the SHA-512 algorithm.
+   * 
+   * @param input The input String to hash.
+   * @return A hashed hexadecimal String of the input.
+   */
+  public static String sha512(String input) {
```

Comment:
SHA-512 hashing is in theory optional, it doesn't provide any security at all because it essentially replaces your password.
If the server never knows the real password, the server can't leak it, so if this password is being used on other sites, they won't be compromised as well.